nimare/meta/utils.py

```diff
@@ -344,7 +347,8 @@ def compute_kda_ma(
         # Use a memmapped 4D array
         kernel_data = np.memmap(memmap_filename, dtype=type(value), mode="w+", shape=kernel_shape)
     else:
-        kernel_data = np.zeros(kernel_shape, dtype=type(value))
+        # Use a list of 3D sparse arrays
+        kernel_data_lst = [None] * n_studies
```

Comment:
If I'm reading correctly the idea here it to use a list of 3D sparse arrays? 

Isn't it possible with `sparse` to construct a 4D array? In a way I believe that's one of the main advantages of `sparse`

Reply:
> If I'm reading correctly the idea here it to use a list of 3D sparse arrays?

Yes, the list of 3D sparse arrays follows the same principle as it was previously implemented. The only advantage is that we avoid the memory peak by accumulating sparse arrays instead of the whole 3D array. But the dense to sparse and sparse to dense formatting is slowing everything.

> Isn't it possible with sparse to construct a 4D array? In a way I believe that's one of the main advantages of sparse

Yes, I think it is possible.

Reply:
I was able to construct the 4D sparse array from coordinates, but this object type `sparse._coo.core.COO` is not iterable. To get this working we would need to modify the `KernelTrnasformer`. Alternatively, we could try again the list of 3D sparse arrays but build the sparse array from coordinates instead of compressing a dense array.